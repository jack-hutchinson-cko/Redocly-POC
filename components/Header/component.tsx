import React from 'react';
import { ThemeToggle } from '../index';
import LoginLink from './LoginLink';
import MenuLink from './MenuLink';
import {
	StyledContainer,
	StyledLink,
	StyledLeftHeader,
	StyledLogo,
	StyledLogoIcon,
	StyledLogoText,
	StyledRightHeader,
} from './styles';

const Header = (): JSX.Element => (
	<StyledContainer>
		<StyledLeftHeader>
			<StyledLogo href="/">
				<StyledLogoIcon />
				<StyledLogoText>Documentation</StyledLogoText>
			</StyledLogo>
			<MenuLink />
		</StyledLeftHeader>
		<StyledRightHeader>
			<LoginLink />
			<StyledLink>Get test account</StyledLink>
			<ThemeToggle />
		</StyledRightHeader>
	</StyledContainer>
);

export default Header;
