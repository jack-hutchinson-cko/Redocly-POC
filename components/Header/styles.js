import { SearchBox } from '@redocly/ui';
import styled from 'styled-components';
import { device } from '../css/tokens';
import { CheckoutLogoIcon } from '../svg';

export const StyledSearchBox = styled(SearchBox)`
	border: 1px solid var(--border);
	border-radius: 8px;
	width: 380px;
	height: 100%;

	input {
		border-radius: 8px;
		color: var(--secondary-font-color);
		margin-top: 5px;
		height: 40px;
		line-height: 24px;
		padding: 16px 8px;
		max-width: 380px;
		width: 380px;
	}

	input::placeholder {
		color: var(--secondary-font-color);
	}

	span {
		top: 45%;
		margin-left: 0px;
	}

	span > span {
		background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' version='1.1' x='0' y='0' viewBox='0 0 490 490' xml:space='preserve'%3E%3Cpath fill='%238688A0' d='M484 455.2L366.2 337.4c29.2-35.6 46.8-81.2 46.8-130.9C413 92.5 301.5 0 206.5 0 92.4 0 0 92.5 0 206.5S92.4 413 206.5 413c49.7 0 95.2-17.5 130.8-46.7L455.1 484c8 8 20.9 8 28.9 0C492 476.1 492 463.1 484 455.2zM206.5 371.9C115.2 371.9 41 297.7 41 206.5S115.2 41 206.5 41C297.7 41 372 115.3 372 206.5S297.7 371.9 206.5 371.9z'/%3E%3C/svg%3E") !important;
	}

	ul {
		background: var(--header-background);
		border-radius: 8px;
		border: 1px solid var(--border);
		top: calc(100% + 15px);
		width: 380px;
	}

	ul div {
		color: var(--font-color);
	}

	ul div:first-child {
		padding-top: 16px;
	}

	${device.toLaptop} {
		width: 280px;

		span {
			margin: auto;
			padding: 0px;
		}

		input {
			width: 280px;
		}
	}

	${device.toTablet} {
		width: 180px;

		span {
			margin: auto;
			padding: 0px;
		}

		input {
			width: 180px;
			margin-top: 4px;
		}

		span span {
			display: none;
		}
	}

	${device.toMobileXL} {
		input {
			height: 32px;
			margin-top: 0px;
		}
	}
`;

export const StyledLogoIcon = styled(CheckoutLogoIcon)`
	padding-top: 6px;
`;

export const StyledContainer = styled.header`
	background: var(--header-background);
	border-bottom: 1px solid var(--border);
	color: var(--font-color);
	display: flex;
	justify-content: space-between;
	padding: 24px 40px;
	width: 100%;
`;

export const StyledLogo = styled.a`
	align-items: center;
	display: flex;
	gap: 12px;
	padding: 0;
	margin: 0;
	text-decoration: none;

	:hover {
		text-decoration: none;
	}
`;

export const StyledLogoText = styled.h3`
	color: var(--font-color);
	font-size: 24px;
	line-height: 40px;
	margin: 0;

	${device.toMobileXL} {
		display: none;
	}
`;

export const StyledLink = styled.a`
	color: var(--font-color);
	cursor: pointer;
	font-size: 16px;
	line-height: 24px;
	letter-spacing: 0em;
	text-align: left;
	margin: auto;
	text-decoration: none;

	:hover {
		text-decoration: none;
	}

	${device.toLaptop} {
		display: none;
	}
`;

export const StyledLeftHeader = styled.div`
	display: flex;
	gap: 40px;

	${device.toMobileXL} {
		gap: 16px;
	}
`;

export const StyledRightHeader = styled.div`
	display: flex;
	gap: 24px;
`;
