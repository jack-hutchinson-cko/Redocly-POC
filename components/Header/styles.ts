import styled from 'styled-components';
import { device } from '../css/tokens';
import { CheckoutLogoIcon } from '../svg';

export const StyledLogoIcon = styled(CheckoutLogoIcon)`
	padding-top: 6px;
`;

export const StyledContainer = styled.header`
	background: var(--header-background);
	border-bottom: 1px solid var(--border);
	color: var(--font-color);
	display: flex;
	justify-content: space-between;
	padding: 24px 40px;
	width: 100%;
`;

export const StyledLogo = styled.a`
	align-items: center;
	display: flex;
	gap: 12px;
	padding: 0;
	margin: 0;
	text-decoration: none;

	:hover {
		text-decoration: none;
	}
`;

export const StyledLogoText = styled.h3`
	color: var(--font-color);
	font-size: 24px;
	line-height: 40px;
	margin: 0;
`;

export const StyledLink = styled.a`
	color: var(--font-color);
	font-size: 16px;
	line-height: 24px;
	letter-spacing: 0em;
	text-align: left;
	margin: auto;

	${device.toTablet} {
		display: none;
	}
`;

export const StyledLeftHeader = styled.div`
	display: flex;
	gap: 136px;

	${device.fromTablet} {
		gap: 40px;
	}
`;

export const StyledRightHeader = styled.div`
	display: flex;
	gap: 24px;
`;
