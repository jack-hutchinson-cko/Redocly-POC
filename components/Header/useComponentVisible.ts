import { useState, useEffect, useRef } from 'react';

export default function useComponentVisible(initialIsVisible: boolean) {
	const [isComponentVisible, setIsComponentVisible] = useState(initialIsVisible);
	const ref = useRef<HTMLDivElement>(null);

	const handleHideDropdown = (event: KeyboardEvent): void => {
		if (event.key === 'Escape') {
			setIsComponentVisible(false);
		}
	};

	const handleClickOutside = (event: Event): void => {
		if (ref.current && !ref.current.contains(event.target as Node)) {
			console.log('clicked outside', ref);
			setIsComponentVisible(false);
		}
	};

	useEffect(() => {
		document.addEventListener('keydown', handleHideDropdown, true);
		document.addEventListener('click', handleClickOutside, true);
		return () => {
			document.removeEventListener('keydown', handleHideDropdown, true);
			document.removeEventListener('click', handleClickOutside, true);
		};
	});

	return { ref, isComponentVisible, setIsComponentVisible };
}
