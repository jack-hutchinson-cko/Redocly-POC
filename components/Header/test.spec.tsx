import { shallow } from 'enzyme';
import React from 'react';
import Header from './component';

describe('Header component', () => {
	test('component matches snapshot', () => {
		expect.assertions(1);

		const wrapper = shallow(<Header />);

		expect(wrapper).toMatchSnapshot();
	});
});
