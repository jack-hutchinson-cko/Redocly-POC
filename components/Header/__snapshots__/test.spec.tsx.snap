// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component component matches snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.a
      href="/"
    >
      <Styled(CheckoutLogoIcon) />
      <styled.h3>
        Documentation
      </styled.h3>
    </styled.a>
    <MenuLink />
  </styled.div>
  <styled.div>
    <LoginLink />
    <styled.a>
      Get test account
    </styled.a>
    <ThemeToggle />
  </styled.div>
</styled.div>
`;
